eng/scripts/Analyze-Code.ps1

```diff
@@ -34,6 +34,19 @@ try {
     if($hasErrors) {
         exit 1
     }
+    
+    # Run tool selection analysis
+    try {
+        & "$PSScriptRoot/Test-ToolSelection.ps1"
+        if ($LASTEXITCODE -ne 0) {
+            Write-Host "❌ Tool selection analysis failed"
+            exit 1
+        }
+    } catch {
+        Write-Host "⚠️  Tool selection analysis encountered an issue: $($_.Exception.Message)"
+        # Don't fail the entire analyze step for tool selection issues
+        Write-Host "Continuing with other analysis steps..."
+    }
```

Comment:
Why does a failure status code result in halting of exaction while an exception continues execution? When both, in theory, would result in an exception status code for this script itself if the exception wasn't caught.

Reply:
You're right, I'll change this to follow the pattern other checks have established already.